spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/transaction/jta/NarayanaJtaConfiguration.java

```diff
@@ -114,10 +115,12 @@ public TransactionManager narayanaTransactionManager() {
 	@Bean
 	@DependsOn("narayanaConfiguration")
 	public RecoveryManagerService narayanaRecoveryManagerService() {
+		RecoveryManager.delayRecoveryManagerThread();
 		return new RecoveryManagerService();
 	}
 
 	@Bean
+	@ConditionalOnMissingBean
```

Comment:
You're correct, that doesn't relate with delaying. However, it does allow application to override recovery bean to have more control. Should I move this annotation to a separate commit?